workflows/index-generation/ncbi-compress/src/ncbi_compress.rs

```diff
@@ -399,44 +440,18 @@ pub mod ncbi_compress {
                     {
                         None
                     } else {
-                        Some((record, hash))
+                        let mut writer = writer.lock().unwrap();
```

Comment:
I would avoid the mutex here and iterate through `unique_in_tree_and_chunk` sequentially and write them. The writes need to happen sequentially anyway so this probably doesn't buy you additional performance and could even be a hindrance.

This is also a bit awkward with `filter_map` which typically doesn't have side effects.

Reply:
true, i moved it to a new block!